File: app/src/main/java/com/example/despensacx/ListaDao.java
package com.example.despensacx;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ListaDao {
    @Insert
    long insert(ListaEntity lista);

    @Update
    void update(ListaEntity lista);

    @Delete
    void delete(ListaEntity lista);

    @Query("SELECT * FROM listas WHERE id = :id LIMIT 1")
    ListaEntity getByIdSync(long id);

    @Query("SELECT * FROM listas WHERE id = :id LIMIT 1")
    LiveData<ListaEntity> getById(long id);

    @Query("SELECT * FROM listas WHERE archivada = 0 ORDER BY id DESC")
    LiveData<List<ListaEntity>> getListasActivas();

    @Query("SELECT * FROM listas WHERE archivada = 1 ORDER BY id DESC")
    LiveData<List<ListaEntity>> getListasArchivadas();

    @Query("SELECT * FROM listas")
    List<ListaEntity> getAllSync();
}

File: app/src/main/kotlin/com/example/despensacx/data/ListaDao.kt
package com.example.despensacx.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface ListaDao {
    @Insert
    fun insert(lista: ListaEntity): Long

    @Update
    fun update(lista: ListaEntity)

    @Delete
    fun delete(lista: ListaEntity)

    @Query("SELECT * FROM listas WHERE id = :id LIMIT 1")
    fun getByIdSync(id: Long): ListaEntity?

    @Query("SELECT * FROM listas WHERE id = :id LIMIT 1")
    fun getById(id: Long): LiveData<ListaEntity>

    @Query("SELECT * FROM listas WHERE archivada = 1 ORDER BY id DESC")
    fun getListasArchivadas(): LiveData<List<ListaEntity>>

    @Query("SELECT * FROM listas")
    fun getAllSync(): List<ListaEntity>

    @Query("SELECT * FROM listas WHERE archivada = 0")
    fun getListasActivasSync(): List<ListaEntity>

    @Query("SELECT * FROM listas WHERE archivada = 0")
    fun getListasActivas(): LiveData<List<ListaEntity>>
}
